src/app/services/request/request-reponse.interface.ts

```diff
@@ -0,0 +1,5 @@
+interface RequestReponse {
+  status: number;
+  response?: any;
+  progress?: number;
```

Comment:
i think that it would be nice if the progress would always return a number, not just sometimes

additionally, you may consider naming it so that it is clear if it is in percent or not. e.g. `progress_in_percent`